tests/integration/test_backup_restore_s3/test.py

```diff
@@ -782,15 +782,24 @@ def test_backup_to_s3_different_credentials(allow_s3_native_copy, use_multipart_
     size = 1000
     if use_multipart_copy:
         size = 10000000
-    (backup_events, _) = check_backup_and_restore(
+    (backup_events, restore_events) = check_backup_and_restore(
         storage_policy,
         backup_destination,
         backup_settings=settings,
         restore_settings=settings,
         size=size
     )
+
     check_system_tables(backup_events["query_id"])
 
+    for events in [backup_events, restore_events]:
+        assert "S3WriteRequestsErrors" not in events
+        assert "DiskS3WriteRequestsErrors" not in events
+        assert "S3ReadRequestsErrors" not in events
+        assert "DiskS3ReadRequestsErrors" not in events
```

Comment:
Those assertions fail without this PR's changes because without this PR's changes it tried native copy first which failed due to different credentials so these metrics increased.